chanzuckerberg/single-cell-data-portal
# chore: Json Logging

- #TICKET_NUMBER

### Reviewers
**Functional:** @atolopko-czi @ebezzi 

**Readability:** @MDunitz 

---


## Changes
- use dictConfig to setup python logging.
- use python-json-logging to convert logs to json.
- log before request and after request.
- add line number, files path, and function name to log message

## QA steps (optional)
1. go to https://jsonlogs-frontend.rdev.single-cell.czi.technology
1. `happy logs jsonlogs backend` to see the json formatted logs.
## Notes for Reviewer
Typical log record output:
```
2022-05-27T12:32:35.017000+00:00 2dc72b2253584184e51f580997e11f1ff24b8b2ce53ba336797e18227317373c { 
"levelname": "INFO", 
"asctime": "2022-06-30T19:49:42.030Z", 
"name": "corpora-api-test", 
"message": "CORS allowed_origins: ['http://.*\\\\.corporanet\\\\.local:\\\\d+', '^http://localhost:\\\\d+']", 
"lineno": 74, 
"funcName": "configure_flask_app", 
"pathname": "/single-cell-data-portal/backend/corpora/api_server/app.py"
}
```
